typescript/core/src/v0/ChunkCursor.ts

```diff
@@ -54,12 +59,23 @@ export class ChunkCursor {
    * and re-sort the cursors.
    */
   compare(other: ChunkCursor): number {
-    // If chunks don't overlap at all, sort earlier chunk first
-    if (this.chunkIndex.messageEndTime < other.chunkIndex.messageStartTime) {
-      return -1;
-    }
-    if (this.chunkIndex.messageStartTime > other.chunkIndex.messageEndTime) {
-      return 1;
+    const reverse = this.reverse;
```

Comment:
What about `other.reverse`?

Reply:
🤔 Good catch but I have no idea. Maybe throw if the two don't match? It's an anachronism in the current IndexedReader implementation to compare two such cursors.

Reply:
I agree, throwing seems reasonable